### DISCUSSION TITLE
Loading locales from JS files over JSON

### DISCUSSION THREAD

**Discussion by mikey-mo** [vinissimus/next-translate#1042] [General] [answered]
Is it possible to load locale from JS over JSON?  I'd rather not decouple key constants and pass strings around in the UI.  JS files give me the ability to set key values as constants which can then be used in components w/o fear of typos, etc.

Anyone have any good suggestions?  Thanks.

**Comment by aralroca** [comment, marked as answer]
yes, you can use any format, you can import in your way using `loadLocalesFrom` config from `i18n.js`, the only thing is to return a Promise with the JSON:

```js
{
  // ...rest of config
  "loadLocaleFrom": (lang, ns) =>
    // You can use a dynamic import, fetch, whatever. You should
    // return a Promise with the JSON file.
    import(`./locales/${lang}/${ns}.js`).then((m) => m.default),
}
```

**Reply by mikey-mo** [reply]
Thanks for the reply -- just realized I should have put this in Discussions rather than Issues.
